lib/Scheduler/sched_region.cpp

```diff
@@ -83,6 +95,27 @@ static bool isBbEnabled(Config &schedIni, Milliseconds rgnTimeout) {
   return true;
 }
 
+static void dumpDDG(DataDepGraph *DDG, llvm::StringRef DDGDumpPath,
+                    llvm::StringRef Suffix = "") {
+  Logger::Info("Writing DDG.");
+  std::string Path = DDGDumpPath;
+  Path += DDG->GetDagID();
+
+  if (!Suffix.empty()) {
+    Path += '.';
+    Path += Suffix;
+  }
+
+  Path += ".ddg";
+  // DagID has a `:` in the name, which symbol is not allowed in a path name.
+  // Replace the `:` with a `.` to produce a legal path name.
+  std::replace(Path.begin(), Path.end(), ':', '.');
+
+  auto f = fopen(Path.c_str(), "w");
+  DDG->WriteToFile(f, RES_SUCCESS, 1, 0);
+  fclose(f);
+}
```

Comment:
This does not handle errors like invalid paths, bad permissions, etc.  fwrite() and fclose() silently fail when the FILE* passed to it is invalid/.  This program will not consequently report any error if there is an error opening the file.

If you want the program to fatally error out if the dump file can't be opened then you should check if `f==NULL` here and then error out here.  If you think it would be better to log the failure and continue then the logging statement should be here.  Either way this is the appropriate place for error handling not on line 59